packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,265 @@
+<script lang="ts" setup>
+import { computed } from "vue";
+
+export type ButtonProps = {
+  label?: string;
+  isLoading?: boolean;
+  isDisabled?: boolean;
+  type?: "button" | "submit" | "reset";
+  color?: "primary" | "secondary" | "danger";
+  variant?: "default" | "outline" | "plain";
+  icon?: string;
+};
+
+const props = withDefaults(defineProps<ButtonProps>(), {
+  label: "",
+  isLoading: false,
+  isDisabled: false,
+  type: "button",
+  color: "primary",
+  variant: "default",
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the button is clicked
+   */
+  click: [];
+}>();
+
+const handleClick = () => {
+  emit("click");
+};
```

Comment:
```suggestion
const handleClick = () => emit("click");
```

You could also inline it directly into the template `@click="emit('click')"`